src/v4/options.rs

```diff
@@ -1014,6 +1141,34 @@ fn decode_inner(
             let name = Name::read(&mut name_decoder)?;
             ClientFQDN(flags, rcode1, rcode2, Domain(name))
         }
+        OptionCode::ClasslessStaticRoute => {
```

Comment:
I think we've got a way to decode this already with `decoder.read_pair_ipv4s()`, `StaticRoutingTable` uses it

Reply:
Again, missed the `Ipv4Net` bit. Just to confirm, I am not familiar with the RFC in detail. How does this look like on the wire? The number of 'd' bits is variable like:

[ prefix | d1 | .. | d4 | r1 | r2 | r3 | r4 ]

Or is there always 4 'd' bits?

Reply:
Yup it is variable. You only have the meaningful bytes for the prefix length.

Basically, if your prefix fits in 2 bytes (so anything from a /9 to a /16), then there will only be 2 `d` bytes. Thus there are between zero (for a `/0`) and four (for `/25` -> `/32`) bytes.

Reply:
Got it. I was playing around with this, I used an internal `Decoder` to track the byte counting. Let me know what you think

Reply:
Ha good idea, I like much more your implementation than mine!

One small thing I'd change though is the handling of `prefix_len` greater than 32. You changed from an explicit "stop parsing" to a `debug_assert!`. I think it would be better to raise an actual error to avoid going on with potentially corrupt data. Moreover, this check is what made `let dest = Ipv4Net::new(dest.into(), prefix_len).unwrap();` unwrap-safe.

Reply:
Good point, do you want to make that change or do you want me to do it? We can introduce a new error type in DecodeError if necessary.

Reply:
> since at least the Decoder::new(decoder.read_slice()) line will grab the entire contents of the option, if the option fails to parse then we have a chance to still decode the rest of the packet

with this behaviour, I think the best course of option is either to abort the parsing of the option or to stop the parsing of the whole packet (rationale being, if one field has a bogus value, it's likely another one is like the option length).

however, `assert!` is not a good idea because it would mean anything parsing dhcp packets could packet with crafted packets. it's fine when you control the clients, but not so much when you don't.

IMHO aborting the current option (simply breaking out of the loop) is enough. send garbage, get garbage. If you're OK with this, I'll do the change.

Reply:
Sounds good.

Reply:
Then it's done.